kernel/kernel-impl/src/main/java/org/sakaiproject/user/impl/BaseUserDirectoryService.java

```diff
@@ -1963,6 +1963,42 @@ public UserEdit newUser(String eid)
 		return u;
 	}
 
+	public boolean updateUserId(String id,String newEmail)
+	{
+		try {
```

Comment:
Actually this only changes the EID, and the security bypass is only for the user ID.